### DISCUSSION TITLE
Getting rid of the closure and repetitive codes

### DISCUSSION THREAD

**Discussion by MojtabaHs** [pointfreeco/swift-composable-architecture#2075] [Q&A]
Hi, oftentimes I face this situation:

```
ForEachStore(
    store.scope(
        state: \.filteredItems,
        action: { Feature.Action.child(.listItem(id: $0.0, action: $0.1)) } // <- 👈 This line
    ),
    content: ListItemView.init
    )
)
```

#### - First question:
How can I get rid of the action closure? Like when it's not nested and we can just call `Feature.Action.childItem`

---

Also, for the scoping in the reducer, I face this:
```
action: (/Action.child).appending(path: /Action.Child.list),
```
It's sort of long and repetitive.

#### - Second question:
Is there any way to make it simpler?

Thanks in advance

**Comment by tgrapperon** [comment]
Hey @MojtabaHs! One alternative question is why is this domain is nested for its action but not its state? You can often rework your model so the child domains are embedded in a "flat" and not "skewed" way, but it can be interesting to know why it's not like that to begin with. Can you tell us more about potential configurations where this happens? Are you using "Action boundaries" (that is sub-actions like `.view(…)`, `.delegate(…)`, etc.)?

**Reply by MojtabaHs** [reply]
I scoped actions because they have different callers. For example, in this case, childs actions could be ignored completely by the parent reducer by just returning `.none` and letting the child reducer does its work. This way I can also put some code-level restrictions on the separated concern.

But about the state, it's actually kind of scoped!, but stored inside (or accessed within) a variable for the sake of usage inside the `viewStore`.
